docs/plans/2026-01-30-extract-extensions-optimization-design.md

```
@@ -0,0 +1,137 @@
+```go
+func ExtractExtensions(data []byte) map[string]any {
+    // Fast path: skip JSON parsing if no extension pattern found
+    if !bytes.Contains(data, []byte(`"x-`)) {
+        return nil
+    }
+    // ... existing implementation
+}
+```
+
+**Results**:
+| Scenario | Before | After | Speedup |
```

markdownlint-cli2 (0.20.0)

[warning] Tables should be surrounded by blank lines

(MD058, blanks-around-tables)